pepys_import/utils/unit_utils.py

```diff
@@ -151,21 +116,43 @@ def distance_between_two_points_haversine(first_location, second_location):
     c = 2 * asin(sqrt(a))
     radius = 6371  # Radius of earth in kilometers. Use 3956 for miles
     distance = c * radius
-    return (
-        (distance * unit_registry.kilometers / unit_registry.hour)
-        .to(unit_registry.meter / unit_registry.second)
-        .magnitude
+    return (distance * unit_registry.kilometers / unit_registry.hour).to(
+        unit_registry.meter / unit_registry.second
     )
 
 
+def convert_distance(distance, units, line_number, errors, error_type):
+    """
+    Converts the given distance value in supplied units to metres formsat
+    :param distance: distance value in string format
+    :type distance: String
+    :param units: units of distance for supplied measurement
+    :type units: String
+    :param line_number: Line number
+    :type line_number: String
+    :param errors: Error List to save value error if it raises
+    :type errors: List
+    :param error_type: Type of error
+    :type error_type: String
+    :return: return the converted speed value
+    """
+    try:
+        valid_distance = float(distance)
+    except ValueError:
+        errors.append(
+            {
+                error_type: f"Line {line_number}. Error in distance value {distance}. "
+                f"Couldn't convert to a number"
+            }
+        )
+        return False
+    distance = (valid_distance * units).to(unit_registry.meter).magnitude
```

Comment:
Hmm, I'd rather this returned a distance with units.  I _don't think_ we need to convert it to metres.  It can move around the codebase as distance with units (`yards`), and get passed into the data_store as that.  Inside the data_store, it can be converted to metres before going into SQL `distance.to(unit_registry.metre).magnitude`.

So, the parsers convert the value read in (distance without units) into a distance with units as soon as possible, and it's that distance with units that gets passed around.

Reply:
Aah, hold on. The above method is only used once, and then it's result is (currently) ignored anyway.  I see we do follow the intended behaviour for `state`.